### PR TITLE
Refactor page filtering

### PR DESCRIPTION
**Pull request by IKarbowiak**
Update methods for filtering pages and page types. 

⚠️ Filtering on `content` will be updated in the separate task: [SALEOR-3458](https://app.clickup.com/t/2549495/SALEOR-3458)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3273 Refactor page filtering.](https://app.clickup.com/t/2549495/SALEOR-3273)

**Review by korycins — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

File: saleor/page/migrations/0023_auto_20210525_1113.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            name="page_meta_idx",
+        ),
+        migrations.RemoveIndex(
+            model_name="pagetype",
+            name="pagetype_p_meta_idx",
+        ),
+        migrations.RemoveIndex(
+            model_name="pagetype",
+            name="pagetype_meta_idx",
+        ),
```
Comment (fowczarek) [inline saleor/page/migrations/0023_auto_20210525_1113.py:29]: We shouldn't remove metadata indexing.

File: saleor/page/models.py [hunk 2]
```diff
@@ -22,6 +23,7 @@ class Page(ModelWithMetadata, SeoModel, PublishableModel):
     class Meta(ModelWithMetadata.Meta):
         ordering = ("slug",)
         permissions = ((PagePermissions.MANAGE_PAGES.codename, "Manage pages."),)
+        indexes = [GinIndex(fields=["title", "slug"])]
```
Comment (fowczarek) [inline saleor/page/models.py:26]: We should remove `ModelWithMetadata.Meta`  from `Meta` inheritance and :
```suggestion
        indexes = [
            *ModelWithMetadata.Meta.indexes,
            GinIndex(fields=["title", "slug"])
        ]
```

File: saleor/page/models.py [hunk 3]
```diff
@@ -64,3 +66,4 @@ class Meta(ModelWithMetadata.Meta):
                 "Manage page types and attributes.",
             ),
         )
+        indexes = [GinIndex(fields=["name", "slug"])]
```
Comment (fowczarek) [inline saleor/page/models.py:69]: Same here.

**Review by fowczarek — CHANGES_REQUESTED** [review]

File: saleor/graphql/page/filters.py [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
 def filter_page_search(qs, _, value):
-    page_fields = ["content", "slug", "title"]
-    qs = filter_by_query_param(qs, value, page_fields)
-    return qs
+    if not value:
+        return qs
+    return qs.filter(
+        Q(title__trigram_similar=value)
+        | Q(slug__trigram_similar=value)
+        | Q(content__icontains=value)
```
Comment (korycins) [inline saleor/graphql/page/filters.py:18]: Is it safe to use `icontains` for page content? Looks like it could be a performance issue.
cc @fowczarek

Comment (fowczarek) [reply saleor/graphql/page/filters.py:18]: We will fix it in a separate task.

File: saleor/graphql/page/filters.py [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
+    )
+
+
+def filter_page_page_types(qs, _, value):
```
Comment (maarcingebala) [inline saleor/graphql/page/filters.py:22]: Where is this function used?

Comment (IKarbowiak) [reply saleor/graphql/page/filters.py:22]: I didn't add the changes in `PageFilter` but it's fixed already.

**Review by fowczarek — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

**Pull request merged**